ros2doctor/ros2doctor/verb/call.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from argparse import ArgumentTypeError
+import socket
+import struct
+import threading
+import time
+
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.node import Node
+from ros2doctor.verb import VerbExtension
+
+from std_msgs.msg import String
+
+DEFAULT_GROUP = '225.0.0.1'
+DEFAULT_PORT = 49150
+
+
+def positive_int(string: str) -> int:
+    try:
+        value = int(string)
+    except ValueError:
+        value = -1
+    if value <= 0:
+        raise ArgumentTypeError('value must be a positive integer')
+    return value
+
+
+class CallVerb(VerbExtension):
+    """
+    Check network connectivity between multiple hosts.
+
+    This command can be invoked on multiple hosts to confirm that they can talk to each other
+    by using talker/listener, multicast send/receive to check topic discovering and
+    UDP communication.
+    This command outputs a summary table of msgs statistics at a custom rate(Hz).
+    """
+
+    def add_arguments(self, parser, cli_name):
+        parser.add_argument(
+            'topic_name', nargs='?', default='/canyouhearme',
+            help="Name of ROS topic to publish to (default: '/canyouhearme')")
+        parser.add_argument(
+            'time_period', nargs='?', default=0.1,
+            help='Time period to publish/send one message (default: 0.1s)')
+        parser.add_argument(
+            'duration', nargs='?', default=20, type=positive_int,
+            help='How long this process runs (default: 20s)')
+        parser.add_argument(
+            '-r', '--rate', metavar='N', type=float, default=1.0,
```

Comment:
I think that "rate" and "period" (or "time_period") may be easily confused. I suggest making this option name more verbose and change it to a period for consistency, e.g. "--output-period".